### PR TITLE
Carry along - Add environment variables documentation (#11550)

### LINKED ISSUE #3294: Document Available Configuration Options
As discussed in #3063 it would be useful to have a list of env vars that can be set. If we implement #2052 (config file support) this would be useful for that as well

### PR DESCRIPTION
**Pull request by kdoh**
Changes from #11550 with edits.
Fixes #3294

### CODE REVIEW

File: docs/operations-guide/environment-variables.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Environment variables
+
+Settings in Metabase are created using a custom macro, `defsetting`. This macro helps to create a Settings model that will be cached and stored in the Database model, viewed/modified from the Admin Panel, and automagically creates a corresponding environment variable that can be used (by default) to override that setting.
```
Comment (tlrobinson) [inline docs/operations-guide/environment-variables.md]: I don't think we need to discuss the `defsetting` implementation detail in docs. Could we just say something like

> Most settings in Metabase can be stored in the application database and viewed/modified in the Admin Panel, or set via environment variables. The environment variables always take precedence over stored settings.

Comment (kdoh) [reply docs/operations-guide/environment-variables.md]: Yeah I like that.

File: docs/operations-guide/environment-variables.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+These environment variables, if defined, will prevent the corresponding setting from being changed in the Admin Panel.
```
Comment (dacort) [inline docs/operations-guide/environment-variables.md]: Is this intended to be a comprehensive list or just the ones defined with `defsetting`? There are some additional settings in `config.clj` and `plugins.clj` (not sure if others exist, I've just used `MB_DB_FILE` and `MB_PLUGIN_DIR` recently) - should we include those as well?

**Review by dacort — CHANGES_REQUESTED** [review]
👍 Nice to have this reference and I think it fits in well with the operational docs.

File: docs/operations-guide/environment-variables.md (hunk 2 above)
Comment (tlrobinson) [reply docs/operations-guide/environment-variables.md]: You're right, there are some cases where we do `(config/config-str :mb-whatever)` without having a corresponding `defsetting`.

Here's a command to get a comprehensive list:

```
(grep -rEoh 'mb-[a-z-]+' src ; grep -rEoh 'defsetting [a-z-]+' src | awk '{ print "mb-" $2 }') | sort | uniq
```

Comment (spdustin) [reply docs/operations-guide/environment-variables.md]: > You're right, there are some cases where we do `(config/config-str :mb-whatever)` without having a corresponding `defsetting`.

Are all those also exposed in the Admin Panel?

**Review by dacort — APPROVED** [review]

File: docs/operations-guide/environment-variables.md (hunk 2 above)
Comment (dacort) [reply docs/operations-guide/environment-variables.md]: Approving for now - there are 119 variables in that list and 55 `defsetting` ones. We can add the other ones as necessary given they may not have built in defaults or descriptions.

**Pull request merged**